launchlib/launcher.go

```diff
@@ -367,6 +405,18 @@ func filterHeapSizeArgsV2(args []string, heapPercentage *float64) ([]string, err
 	return filtered, nil
 }
 
+func getNativeArgsFromJVMOpts(jvmOpts []string) []string {
+	var nativeArgs []string
+	for _, arg := range jvmOpts {
+		for _, allowed := range AllowedNativeImageJVMOptions {
+			if strings.HasPrefix(arg, allowed) {
+				nativeArgs = append(nativeArgs, arg)
+			}
+		}
+	}
+	return nativeArgs
+}
```

Comment:
Thoughts on putting this and the whitelist into a separate file? (I imagine the whitelist will grow over time and it might be beneficial for it to live separately, and will make it easier to realize that it's only based on prefix matching, which will require some changes if we want certain options)

Reply:
Moved some native logic over to native.go